src/installer/tests/Microsoft.NET.HostModel.Tests/Helpers/BundleHelper.cs

```diff
@@ -57,17 +58,35 @@ public static string[] GetBundledFiles(TestProjectFixture fixture)
             return new string[] { $"{appBaseName}.dll", $"{appBaseName}.deps.json", $"{appBaseName}.runtimeconfig.json" };
         }
 
-        public static string[] GetExtractedFiles(TestProjectFixture fixture)
+        public static string[] GetExtractedFiles(TestProjectFixture fixture, BundleOptions bundleOptions)
         {
-            return new string[] { Path.GetFileName(fixture.TestProject.CoreClrDll) };
+            switch (bundleOptions)
+            {
+                case BundleOptions.None:
+                case BundleOptions.BundleOtherFiles:
+                case BundleOptions.BundleSymbolFiles:
+                    throw new ArgumentException($"Bundle option {bundleOptions} doesn't extract any files to disk.");
+
+                case BundleOptions.BundleAllContent:
+                    return Directory.GetFiles(Path.Combine(fixture.TestProject.OutputDirectory, fixture.RepoDirProvider.Configuration, fixture.Framework, fixture.CurrentRid))
+                        .Select(f => Path.GetFileName(f))
+                        .Except(GetFilesNeverExtracted(fixture)).ToArray();
+
+                case BundleOptions.BundleNativeBinaries:
+                    return new string[] { Path.GetFileName(fixture.TestProject.CoreClrDll) };
+
+                default:
+                    throw new ArgumentException("Unsupported bundle option.");
+            }
         }
 
         public static string[] GetFilesNeverExtracted(TestProjectFixture fixture)
         {
             string appBaseName = GetAppBaseName(fixture);
-            return new string[] { $"{appBaseName}.deps.json",
-                                  $"{appBaseName}.runtimeconfig.json",
-                                  $"{appBaseName}.dll",
+            return new string[] { $"{appBaseName}.dll",
+                                  $"{appBaseName}.exe",
+                                  $"{appBaseName}.pdb",
+                                  $"{appBaseName}.runtimeconfig.dev.json",
```

Comment:
I would expect this to break on non-Windows - we need the `{appBaseName}` (without extension) since that's the executable on non-Windows.